protobufs/src/cdh.cpp

```diff
@@ -0,0 +1,77 @@
+#include "cdh.hpp"
+
+using namespace std;
+
+typedef bool (*collect_telemetry_function)(cdh::telemetry::TMFrame &frame);
+
+namespace cdh {
+namespace subsystems {
+uint seq_no = 0;
+
+cdh::telemetry::TMFrame collect_all_telemetry() {
+  // TODO: Move this out of the function.
+  collect_telemetry_function funcs[] = {
+      pwr::collect_telemetry, imu::collect_telemetry, hmi::collect_telemetry};
+
+  cdh::telemetry::TMFrame frame;
+  // Add the management TM point(s) first
+  // TM ID 01 -- Max TM frame sequence number
+  cdh::telemetry::Telemetry *tm = frame.add_tm();
+  tm->set_sys(CDH);
+  tm->set_id(0x01);
+  tm->set_int_value(UINT_MAX);
+  // Get the telemetry from all the subsystems
+  for (int i = 0; i < 3; ++i) {
+    if (!funcs[i](frame)) {
+      // TODO: A better error printing statement.
```

Comment:
On Zephyr this might be replaced with a `printk` to print to the kernel journal.